src/Solve.jl

```diff
@@ -97,6 +97,7 @@ function matrix_normal_form_type(R::Ring)
 end
 
 matrix_normal_form_type(::Field) = RREFTrait()
+matrix_normal_form_type(::Matrix{Rational{BigInt}}) = FFLUTrait()
```

Comment:
Should this be the type of the field?

Reply:
The reason for the (still) failing tests should be that `MatRingElem` is not a subtype of `MatElem`, so all the `can_solve_...` methods don't apply.